### PR TITLE
Fix hostcfgd crash when delete entire config table.

### PR DESCRIPTION
**Pull request by liuh-80**
**What I did**
Fix hostcfgd crash when delete entire config table.

**Why I did it**
hostcfgd subscribe table change in register_callbacks() method. When a config table been deleted, the 'data' parameter of callback method will be 'None', however most callback doesn't handle the 'None' case, they only handle empty dictionary case. when this happen hostcfgd will crash.

**How I verified it**
Pass all UT.
Add new UT for code coverage.

##### Work item tracking
- Microsoft ADO: 27166522

**Details if related**

### CODE REVIEW

**Comment by qiluo-msft** [comment]
Could you paste the crash stack? Could you check the coverage?

File: scripts/hostcfgd [hunk 1]
```diff
@@ -1796,5 +1797,9 @@ def main():
     daemon.start()
 
 if __name__ == "__main__":
-    main()
+    try:
+        main()
+    except Exception as e:
```
Comment (qiluo-msft) [inline scripts/hostcfgd:1802]: >except [](http://example.com/codeflow?start=4&length=6)

This bugfix is not related to your main topic, we can move it to another PR. And this is not the ideal way to fix a generic issue relating systemd handle simple service return code. 
Not sure if this help [https://unix.stackexchange.com/a/734184](https://unix.stackexchange.com/a/734184)

**Comment by qiluo-msft** [comment]
@alexrallen @dgsudharsan Please help review this PR? wondering why this is not identified by original testcases?

**Comment by liuh-80** [comment]
> Could you paste the crash stack? Could you check the coverage?

Here is the crash stack in syslog:

Mar 13 06:40:24.354944 vlab-01 INFO hostcfgd[17776]: Traceback (most recent call last):
Mar 13 06:40:24.355016 vlab-01 INFO hostcfgd[17776]:   File "/usr/local/bin/hostcfgd", line 1799, in <module>
Mar 13 06:40:24.355353 vlab-01 INFO hostcfgd[17776]:     main()
Mar 13 06:40:24.355403 vlab-01 INFO hostcfgd[17776]:   File "/usr/local/bin/hostcfgd", line 1796, in main
Mar 13 06:40:24.355596 vlab-01 INFO hostcfgd[17776]:     daemon.start()
Mar 13 06:40:24.355643 vlab-01 INFO hostcfgd[17776]:   File "/usr/local/bin/hostcfgd", line 1788, in start
Mar 13 06:40:24.355975 vlab-01 INFO hostcfgd[17776]:     self.config_db.listen(init_data_handler=self.load)
Mar 13 06:40:24.356032 vlab-01 INFO hostcfgd[17776]:   File "/usr/lib/python3/dist-packages/swsscommon/swsscommon.py", line 2539, in listen
Mar 13 06:40:24.356884 vlab-01 INFO hostcfgd[17776]:     self.__fire(table, row, data)
Mar 13 06:40:24.356955 vlab-01 INFO hostcfgd[17776]:   File "/usr/lib/python3/dist-packages/swsscommon/swsscommon.py", line 2606, in __fire
Mar 13 06:40:24.357448 vlab-01 INFO hostcfgd[17776]:     handler(table, key, data)
Mar 13 06:40:24.357505 vlab-01 INFO hostcfgd[17776]:   File "/usr/local/bin/hostcfgd", line 1733, in callback
Mar 13 06:40:24.357844 vlab-01 INFO hostcfgd[17776]:     return func(key, op, data)
Mar 13 06:40:24.357899 vlab-01 INFO hostcfgd[17776]:            ^^^^^^^^^^^^^^^^^^^
Mar 13 06:40:24.357940 vlab-01 INFO hostcfgd[17776]:   File "/usr/local/bin/hostcfgd", line 1599, in aaa_handler
Mar 13 06:40:24.358215 vlab-01 INFO hostcfgd[17776]:     self.aaacfg.aaa_update(key, data)
Mar 13 06:40:24.358263 vlab-01 INFO hostcfgd[17776]:   File "/usr/local/bin/hostcfgd", line 304, in aaa_update
Mar 13 06:40:24.358430 vlab-01 INFO hostcfgd[17776]:     self.modify_conf_file()
Mar 13 06:40:24.358472 vlab-01 INFO hostcfgd[17776]:   File "/usr/local/bin/hostcfgd", line 487, in modify_conf_file
Mar 13 06:40:24.358523 vlab-01 INFO hostcfgd[17776]:     authorization.update(self.authorization)
Mar 13 06:40:24.358581 vlab-01 INFO hostcfgd[17776]: TypeError: 'NoneType' object is not iterable
Mar 13 06:40:24.540918 vlab-01 NOTICE systemd[1]: hostcfgd.service: Main process exited, code=exited, status=1/FAILURE

File: scripts/hostcfgd (hunk 1 above)
Comment (liuh-80) [reply scripts/hostcfgd]: Fixed, change reverted

**Review by qiluo-msft — APPROVED** [review]
